agent/engine/task_manager.go

```diff
@@ -802,6 +902,45 @@ func (mtask *managedTask) containerNextState(container *api.Container) *containe
 	}
 }
 
+func (mtask *managedTask) resourceNextState(resource taskresource.TaskResource) *resourceTransition {
+	resKnownStatus := resource.GetKnownStatus()
+	resDesiredStatus := resource.GetDesiredStatus()
+
+	if resKnownStatus == resDesiredStatus {
+		seelog.Debugf("Managed task [%s]: resource [%s] at desired status: %s",
+			mtask.Arn, resource.GetName(), resDesiredStatus.String())
+		return &resourceTransition{
+			nextState:      taskresource.ResourceStatus(0), // set to none
```

Comment:
nit: should we create types for these resource status? also i thought we already had done that hm

Reply:
Each resource has its own status type which is of the type `taskresource.ResourceStatus`. Hence using a generic status type might not solve the problem here.